src/test/java/com/teragrep/pth10/TeragrepTransformationTest.java

```diff
@@ -581,4 +583,100 @@ public void tgGetArchiveSummaryTest() {
+                            Assertions.assertTrue(result.stream().allMatch(s -> pattern.matcher(s).matches()));
+                            Assertions.assertTrue(result.stream().allMatch(s -> s.equals("127") || s.equals("4")));
+                        }
+                );
+    }
+
+    @Test
+    @DisabledIfSystemProperty(
+            named = "skipSparkTest",
+            matches = "true"
+    )
+    public void tgRegexExctractDefaultInputTest() {
+        // default "_raw": "5"
+        String regex = "\\d+";
+        Pattern pattern = Pattern.compile(regex);
+        streamingTestUtil
+                .performDPLTest(
+                        "index=index_A | teragrep exec regexextract regex " + regex + " output strTokens", testFile,
+                        ds -> {
+                            List<String> result = ds
+                                    .select("strTokens")
+                                    .first()
+                                    .getList(0)
+                                    .stream()
+                                    .map(Object::toString)
+                                    .collect(Collectors.toList());
+                            Assertions.assertEquals(1, result.size());
+                            Assertions.assertTrue(result.stream().allMatch(s -> pattern.matcher(s).matches()));
+                            Assertions.assertTrue(result.stream().allMatch(s -> s.equals("5")));
+                        }
+                );
+    }
+
+    @Test
+    @DisabledIfSystemProperty(
+            named = "skipSparkTest",
+            matches = "true"
+    )
+    public void tgRegexExtractDefaultOutputTest() {
+        // source: "127.4.4.4"
+        String regex = "\\d+";
+        Pattern pattern = Pattern.compile(regex);
+        streamingTestUtil
+                .performDPLTest(
+                        "index=index_A | teragrep exec regexextract regex " + regex + " input source", testFile, ds -> {
+                            List<String> result = ds
+                                    .select("tokens")
+                                    .first()
+                                    .getList(0)
+                                    .stream()
+                                    .map(Object::toString)
+                                    .collect(Collectors.toList());
+                            Assertions.assertEquals(4, result.size());
+                            Assertions.assertTrue(result.stream().allMatch(s -> pattern.matcher(s).matches()));
+                            Assertions.assertTrue(result.stream().allMatch(s -> s.equals("127") || s.equals("4")));
+                        }
+                );
+    }
+
+    @Test
+    @DisabledIfSystemProperty(
+            named = "skipSparkTest",
+            matches = "true"
+    )
+    public void tgRegexExtractionMissingRegexExceptionTest() {
+        Assertions.assertThrows(AssertionFailedError.class, () -> {
```

Comment:
Is it possible to assert the exception message here as well?

Reply:
Added check for exception message